From 544375356f0654bd8a9068624ee08219fa9cd76e Mon Sep 17 00:00:00 2001
From: Yuta Nakai <nak5124@live.jp>
Date: Mon, 16 Sep 2024 20:53:49 +0900
Subject: [PATCH 14/15] mingw-w64: When building executables, not DLLs, add
 --tsaware.

---
 gcc/config/i386/mingw-w64.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/i386/mingw-w64.h b/gcc/config/i386/mingw-w64.h
index 06253e9fb11..951543438b7 100644
--- a/gcc/config/i386/mingw-w64.h
+++ b/gcc/config/i386/mingw-w64.h
@@ -138,7 +138,7 @@ along with GCC; see the file COPYING3.  If not see
   %{shared|mdll: " SUB_LINK_ENTRY " --enable-auto-image-base} \
   " LINK_SPEC_LARGE_ADDR_AWARE "\
   " LINK_SPEC_DISABLE_DYNAMICBASE "\
-  %(shared_libgcc_undefs)"
+  %(shared_libgcc_undefs) %{!shared: %{!mdll: --tsaware}}"
 
 /* Enable sincos optimization, overriding cygming.h.  sincos, sincosf
    and sincosl functions are available on mingw-w64, but not on the
-- 
2.46.1

