diff --git a/contrib/seekable_format/zstdseek_decompress.c b/contrib/seekable_format/zstdseek_decompress.c
index 4fefc5f2..d754e3d7 100644
--- a/contrib/seekable_format/zstdseek_decompress.c
+++ b/contrib/seekable_format/zstdseek_decompress.c
@@ -77,7 +77,7 @@
 /* ************************************************************
 * Avoid fseek()'s 2GiB barrier with MSVC, macOS, *BSD, MinGW
 ***************************************************************/
-#if defined(_MSC_VER) && _MSC_VER >= 1400
+#if (defined(_MSC_VER) && _MSC_VER >= 1400) || defined(__MINGW32__)
 #   define LONG_SEEK _fseeki64
 #elif !defined(__64BIT__) && (PLATFORM_POSIX_VERSION >= 200112L) /* No point defining Large file for 64 bit */
 #   define LONG_SEEK fseeko
diff --git a/programs/fileio_common.h b/programs/fileio_common.h
index 55491b8e..d64a7fe9 100644
--- a/programs/fileio_common.h
+++ b/programs/fileio_common.h
@@ -80,7 +80,7 @@ extern UTIL_time_t g_displayClock;
 
 
 /* Avoid fseek()'s 2GiB barrier with MSVC, macOS, *BSD, MinGW */
-#if defined(_MSC_VER) && _MSC_VER >= 1400
+#if (defined(_MSC_VER) && _MSC_VER >= 1400) || defined(__MINGW32__)
 #   define LONG_SEEK _fseeki64
 #   define LONG_TELL _ftelli64
 #elif !defined(__64BIT__) && (PLATFORM_POSIX_VERSION >= 200112L) /* No point defining Large file for 64 bit */
diff --git a/programs/util.c b/programs/util.c
index 7f65f937..09db4670 100644
--- a/programs/util.c
+++ b/programs/util.c
@@ -173,7 +173,7 @@ int UTIL_fstat(const int fd, const char* filename, stat_t* statbuf)
 {
     int ret;
     UTIL_TRACE_CALL("UTIL_stat(%d, %s)", fd, filename);
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) || defined(__MINGW32__)
     if (fd >= 0) {
         ret = !_fstat64(fd, statbuf);
     } else {
diff --git a/programs/util.h b/programs/util.h
index 571d3942..0a26b7b0 100644
--- a/programs/util.h
+++ b/programs/util.h
@@ -29,7 +29,7 @@ extern "C" {
 /*-************************************************************
 * Avoid fseek()'s 2GiB barrier with MSVC, macOS, *BSD, MinGW
 ***************************************************************/
-#if defined(_MSC_VER) && (_MSC_VER >= 1400)
+#if (defined(_MSC_VER) && (_MSC_VER >= 1400)) || defined(__MINGW32__)
 #  define UTIL_fseek _fseeki64
 #elif !defined(__64BIT__) && (PLATFORM_POSIX_VERSION >= 200112L) /* No point defining Large file for 64 bit */
 #  define UTIL_fseek fseeko
@@ -108,7 +108,7 @@ int UTIL_requireUserConfirmation(const char* prompt, const char* abortMsg, const
     typedef struct __stat64 stat_t;
     typedef int mode_t;
 #elif defined(__MINGW32__) && defined (__MSVCRT__)
-    typedef struct _stati64 stat_t;
+    typedef struct __stat64 stat_t;
 #else
     typedef struct stat stat_t;
 #endif
diff --git a/zlibWrapper/gzlib.c b/zlibWrapper/gzlib.c
index c7265151..209f5744 100644
--- a/zlibWrapper/gzlib.c
+++ b/zlibWrapper/gzlib.c
@@ -8,7 +8,7 @@
 
 #include "gzguts.h"
 
-#if defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
+#if defined(_WIN32) && !defined(__BORLANDC__)
 #  define LSEEK _lseeki64
 #else
 #if defined(_LARGEFILE64_SOURCE) && _LFS64_LARGEFILE-0
